src/UsgsAstroLsSensorModel.cpp

```diff
@@ -1938,6 +1941,25 @@ void UsgsAstroLsSensorModel::getAdjSensorPosVel(const double& time,
   double sensPosNom[3];
   lagrangeInterp(m_numPositions / 3, &m_positions[0], m_t0Ephem, m_dtEphem,
                  time, 3, nOrder, sensPosNom);
+
+  // Avoid computing the velocity and adjustments, if not needed, as those
+  // take up at least half of this function's time.
+  bool has_adj = false;
+  for (size_t it = 0; it < adj.size(); it++) {
+    if (adj[it] != 0) {
+      has_adj = true;
+    }
+  }
```

Comment:
This can be replaced with `bool has_dj = std::all_of(adj.begin(), adj.end(), [](double val){return val == 0.0});` I imagine the compiler will optimize both of these implementations away though.

Reply:
I think that lambda functions are kind of terrible for readability. They have their uses, such as being able to quickly cook up a function object to pass to another function, but in this case the simple loop checking for 0 seems to be much easier to understand than this. And yeah, under the hood likely the performance is the same.